crates/repo/src/lib.rs

```diff
@@ -108,6 +194,26 @@ pub fn app() -> Router {
                         "/_tree",
                         tree::app(trees.entry(namespace.clone()).or_default()),
                     )
```

Comment:
We need to add a `tower::Layer` here, which would check that a repository identified by `namespace` exists before calling these routes